### PR TITLE
Week #26

### PR DESCRIPTION
**Pull request by cmauf**

### CODE REVIEW

**Comment by Laudablerug** [comment]
Course & CourseHeader have been documented.

File: docs/components/course/course-header.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+This component is used to display the header of the course detail page. ut displays the course title, the breadcrumb and the title of the content to display. 
+
+## Structure
+
+### Dependencies
+* `slugify`,`unslugify` from `@/mixins/general/course-structure`
+
+### Mixins
+* locale
```
Comment (cmauf) [inline docs/components/course/course-header.md]: backticks missing

File: docs/components/course/course-header.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Component Functionality
+---------
+
+### Props
+- `contentTitle`: it expects an object to be passed 
+- `courseName`: expects a string to be passed
+- `coursePath`: expects a string to be passed
+
```
Comment (cmauf) [inline docs/components/course/course-header.md]: mention if required or not

File: docs/components/course/course-header.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+CourseHeader
+===============
+
+This component is used to display the header of the course detail page. ut displays the course title, the breadcrumb and the title of the content to display. 
+
+## Structure
+
```
Comment (cmauf) [inline docs/components/course/course-header.md:7]: missing parent

File: docs/components/course/course-header.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+* `slugify`,`unslugify` from `@/mixins/general/course-structure`
+
+### Mixins
+* locale
+
+Component Functionality
+---------
+
+### Props
+- `contentTitle`: it expects an object to be passed
```
Comment (cmauf) [inline docs/components/course/course-header.md]: suggestion: "- `contentTitle`: required, it expects an object to be passed"

Comment (cmauf) [reply docs/components/course/course-header.md]: in contrast to non-required, where we would put "standard value: XYZ"

File: docs/components/course/course.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- `lyScrollToTop`from `@/components/helpers/scroll-to-top.vue`
+- `CourseEdit` from `@/components/course/course-edit/course-edit.vue`
+- `CourseHeader`
+
+### Mixins
+
+* locale
+* routes
+* storeHandler
+
```
Comment (cmauf) [inline docs/components/course/course.md]: missing backticks

File: docs/components/course/course.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+#### Local
+- `isCourseAuthor`: return true if logged user is admin or author of the course
+- `onFinishDummy`: returns empty function on every [] invocation, is used as a fallback when a real function is not available.
+- `contentToDisplay`: return current content object
+- `viewPermit`: returns true if user is allowed to see selected course 
+
+### Methods
+- `getCourse`: get course from backend, set title 
+- `followingContent`: returns follow set for content block ?(followingContent not followContent?)
```
Comment (cmauf) [inline docs/components/course/course.md]: move to bottom

File: docs/components/course/course.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+### Watchers
+- `courseFlags`: update when courseFlags change
+ 
+### Lifecycle Events
+
+#### created
+- call `getCourse`
+- call `enrollmentFetch`
```
Comment (cmauf) [inline docs/components/course/course.md:62]: before watchers

**Review by cmauf — COMMENTED** [review]
small changes to fix

File: docs/components/course/course.md (hunk 7 above)
Comment (Laudablerug) [reply docs/components/course/course.md:62]: but created under ### Lifecycle Events!

File: docs/components/course/course-edit/course-block-delete.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+
+Component Functionality
+---------
+
+### Computed Properties
+#### VueX Getters
+- `courseContent`
+
+#### Local
+- `contentToDisplay`: return current content block
```
Comment (cmauf) [inline docs/components/course/course-edit/course-block-delete.md]: moved to mixin, therefore no longer there

File: docs/components/course/course-edit/course-block-edit.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+Component Functionality
+---------
+
+### Computed Properties
+#### VueX Getters
+- `courseContent`
+
+#### Local
+- `contentToDisplay`: return current content object
```
Comment (cmauf) [inline docs/components/course/course-edit/course-block-edit.md]: ditto

File: docs/components/course/course-edit/course-block-delete.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+===============
+This component provides tool to delete content block
+
+## Structure
+
+### Roots
+* `CourseEditTools` 
+
+### Dependencies
+* `vuex`
```
Comment (cmauf) [inline docs/components/course/course-edit/course-block-delete.md]: no longer needed

File: docs/components/course/course-edit/course-block-edit.md [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+===============
+Reroute to edit content block 
+
+## Structure
+
+### Roots
+* `CourseEditTools`
+                
+### Dependencies
+* `vuex`
```
Comment (cmauf) [inline docs/components/course/course-edit/course-block-edit.md]: no longer needed

File: docs/components/course/course-edit/course-category-change.md [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+### Data
+- `badInput`: is set to null to track whether the user has entered invalid input
+- `newCategory`:  is initialized as an empty string to store the user's input for the new category.
+
+### Computed Properties
+#### VueX Getters
+- `course`
+
+#### Local
+- `inputWarning`: returns a string value, which is either the localized error message corresponding to the "badInput" property, or a default message if "badInput" is null or undefined. This computed property displays a warning message to the user if they entered invalid input.
```
Comment (cmauf) [inline docs/components/course/course-edit/course-category-change.md]: somehow redundant, also missing backticks for properties

File: docs/components/course/course-edit/course-category-change.md [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+### Methods
+- `changeCourseCategory`: change course category
```
Comment (cmauf) [inline docs/components/course/course-edit/course-category-change.md]: the method also checks for empty or duplicate category first

File: docs/components/course/course-edit/course-content.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+- `courseContent`: 
+
+#### Local
+- `cid`: returns the type of content
+- `comps`: return create, edit, view components for given content type
+- `editContent`: returns true if it's editing an existing content 
+- `stepData`: return input data for content 
+
+### Watchers
+- `saveChange`:is set to true, it calls the "save" method to save the edited content.
```
Comment (cmauf) [inline docs/components/course/course-edit/course-content.md]: a bit misleading: Would rephrase to "when set to true...", also missing backticks

File: docs/components/course/course-edit/course-content.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+### Lifecycle Events
+#### mounted
+- `!this.cid && !this.content`:called when the component is mounted and checks if the necessary data has been passed to the component. If the data is missing, it navigates back to the previous page using the Vue Router's "back" method.
+
+#### beforeDestroy
+- `this.editContent`:calls the "save" method and dispatches an action to update the course
```
Comment (cmauf) [inline docs/components/course/course-edit/course-content.md]: rephrase: It does that when editContent is ture

File: docs/components/course/course-edit/course-content.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+- `comps`: return create, edit, view components for given content type
+- `editContent`: returns true if it's editing an existing content 
+- `stepData`: return input data for content 
+
+### Watchers
+- `saveChange`:is set to true, it calls the "save" method to save the edited content.
+
+### Lifecycle Events
+#### mounted
+- `!this.cid && !this.content`:called when the component is mounted and checks if the necessary data has been passed to the component. If the data is missing, it navigates back to the previous page using the Vue Router's "back" method.
```
Comment (cmauf) [inline docs/components/course/course-edit/course-content.md]: lose everything before "checks"

File: docs/components/course/course-edit/course-edit-header.md [hunk 17]
```diff
@@ -0,0 +1,27 @@
+CourseEditHeader
+===============
+This component provides navigation buttons on top of edit tools 
+
+## Structure
+
```
Comment (cmauf) [inline docs/components/course/course-edit/course-edit-header.md:6]: Roots missing

File: docs/components/course/course.md [hunk 18]
```diff
@@ -53,16 +53,15 @@ Component Functionality
 - `contentToDisplay`: return current content object
 - `viewPermit`: returns true if user is allowed to see selected course
```
Comment (cmauf) [inline docs/components/course/course.md:54]: `followContent`is missing

File: docs/components/course/course-edit/course-edit.md [hunk 19]
```diff
@@ -0,0 +1,26 @@
+CourseEdit
+===============
+This component wraps all course edit tools.
+
+## Structure
+
+### Dependencies
+* `CourseEditHeader`from `@/components/course/course-edit/course-edit-header.vue`
+* `CourseEditTools` from `@/components/course/course-edit/course-edit-tools.vue`
```
Comment (cmauf) [inline docs/components/course/course-edit/course-edit.md]: redundant

File: docs/components/course/course.md [hunk 20]
```diff
@@ -17,9 +17,9 @@ Show Course Content
```
Comment (cmauf) [inline docs/components/course/course.md:15]: ll 14-15 are lazy loaded

**Review by cmauf — COMMENTED** [review]
some fixes needed

**Comment by cmauf** [comment]
@Laudablerug you resolved the conversations, but did not push your changes

**Review by cmauf — COMMENTED** [review]
ok

**Comment by Laudablerug** [comment]
mapGetters(['courseContent']) c here in mapGetters is written small?
